common/lib/protocol-definitions/src/clients.ts

```diff
@@ -5,27 +5,76 @@
 
 import { IUser } from "./users";
 
+/**
+ * A client's connection mode. Readonly or read/write.
+ */
 export type ConnectionMode = "write" | "read";
 
+/**
+ * TODO
+ */
 export interface ICapabilities {
+	/**
+	 * TODO
+	 */
 	interactive: boolean;
 }
+
+/**
+ * TODO
+ */
 export interface IClientDetails {
+	/**
+	 * TODO
+	 */
 	capabilities: ICapabilities;
+	
+	/**
+	 * TODO
+	 */
 	type?: string;
+
 	/**
 	 * If the environment needs to specify multiple properties which gives info about the environment, then
-	 * it should be in particular format like: "prop1:val1;prop2:val2;prop3:val3"
+	 * it should be in particular format like: `prop1:val1;prop2:val2;prop3:val3`.
 	 */
 	environment?: string;
+	
+	/**
+	 * TODO
+	 */
 	device?: string;
 }
 
+/**
+ * TODO
+ */
 export interface IClient {
+	/**
+	 * {@inheritDoc ConnectionMode}
+	 */
 	mode: ConnectionMode;
+	
+	/**
+	 * TODO
+	 */
 	details: IClientDetails;
+	
+	/**
+	 * TODO
+	 */
 	permission: string[];
+	
+	/**
+	 * The user associated with this client.
+	 * 
+	 * @remarks There may be more than 1 client associated with the same user.
+	 */
 	user: IUser;
+	
+	/**
+	 * TODO
+	 */
 	scopes: string[];
```

Comment:
I don't know the details or full set of possible values, but here's what I've seen when inspecting the initial ops of a connection. The service seems to be the one to return these, not the client ask for them (at least not in the session I was looking at):

<img width="329" alt="image" src="https://user-images.githubusercontent.com/716334/228086526-8baa419f-f6c5-42aa-b8d2-11f9447924fe.png">

Reply:
It would be nice to have a concise list to point to. If this is a string array for forwards/backwards compatibility reasons, it would still be nice to document the current values expected by this library.

Additionally, what are consumer responsibilities for the known cases? And what should they do if they encounter an unexpected value here?